src/v/kafka/server/handlers/topics/types.h

```diff
@@ -136,8 +136,13 @@ struct topic_op_result {
 };
 
 inline creatable_topic_result
-from_cluster_topic_result(const cluster::topic_result& err) {
-    return {.name = err.tp_ns.tp, .error_code = map_topic_error_code(err.ec)};
+from_cluster_topic_result(const cluster::topic_result& res) {
```

Comment:
One other thought, does it matter that we would be making this change in a minor release? Normally these changes go to dev where the versions of the structs would increase with major releases

Reply:
Incidentally, this change seems to be breaking a controller upgrade ducktape test, so that sort of supports your concern. I'll poke around a bit, but it seems like this will require some additional care.

If we take your suggested approach of just returning whatever values we asked for (on success), probably we can do all that in the kafka layer and there's no need to send anything new up from the controller, right?

Reply:
> If we take your suggested approach of just returning whatever values we asked for (on success), probably we can do all that in the kafka layer and there's no need to send anything new up from the controller, right?

Yeah I agree +1